scripts/update_categories.py

```
@@ -0,0 +1,61 @@
+FILTER_RE = re.compile(r"\^.*$")
+
+
+def _parse_lines(lines: Iterable[str]) -> list[str]:
+    """Parse host rules from a list of lines."""
+    hosts: list[str] = []
+    for line in lines:
+        line = line.strip()
+        if not line or line.startswith("#") or line.startswith("!"):
+            continue
+        line = HOST_RE.sub("", line)
+        line = line.lstrip("||")
```

Ruff (0.11.9)

Using `.strip()` with multi-character strings is misleading

(B005)